packages/graz/src/hooks/methods.ts

```diff
@@ -99,3 +115,191 @@ export const useSendIbcTokens = ({
+ * })
+ *
+ * const instantiateMessage = { foo: 'bar' };
+ * instantiateMyContract(instantiateMessage);
+ * ```
+ */
+export const useInstantiateContract = <Message extends Record<string, unknown>>({
+  codeId,
+  onError,
+  onLoading,
+  onSuccess,
+}: UseInstantiateContractArgs<Message>) => {
+  const { data: account } = useAccount();
+  const accountAddress = account?.bech32Address;
+
+  const mutationFn = (args: InstantiateContractMutationArgs<Message>) => {
+    const contractArgs: InstantiateContractArgs<Message> = {
+      ...args,
+      fee: args.fee ?? "auto",
+      senderAddress: accountAddress,
+      codeId,
+    };
+
+    return instantiateContract(contractArgs);
+  };
+
+  const queryKey = ["USE_INSTANTIATE_CONTRACT", onError, onLoading, onSuccess, codeId];
+  const mutation = useMutation(queryKey, mutationFn, {
+    onError: (err, data) => Promise.resolve(onError?.(err, data)),
+    onMutate: onLoading,
+    onSuccess: (txResponse) => Promise.resolve(onSuccess?.(txResponse)),
+  });
+
+  return {
+    error: mutation.error,
+    isLoading: mutation.isLoading,
+    isSuccess: mutation.isSuccess,
+    instantiateContract: mutation.mutate,
+    instantiateContractAsync: mutation.mutateAsync,
+    status: mutation.status,
+  };
+};
+
+export type UseExecuteContractArgs<Message extends Record<string, unknown>> = {
+  contractAddress: string;
+} & MutationEventArgs<ExecuteContractMutationArgs<Message>, ExecuteResult>;
+
+/**
+ * graz mutation hook for executing transactions against a CosmWasm smart
+ * contract.
+ *
+ * @example
+ * ```ts
+ * import { useExecuteContract } from "graz"
+ *
+ * interface GreetMessage {
+ *   name: string;
+ * }
+ *
+ * interface GreetResponse {
+ *   message: string;
+ * }
+ *
+ * const contractAddress = "cosmosfoobarbaz";
+ * const { executeContract } = useExecuteContract<ExecuteMessage>({ contractAddress });
+ * executeContract({ name: 'CosmWasm' }, {
+ *   onSuccess: (data: GreetResponse) => console.log('Got message:', data.message);
+ * });
+ * ```
+ */
+export const useExecuteContract = <Message extends Record<string, unknown>>({
+  contractAddress,
+  onError,
+  onLoading,
+  onSuccess,
+}: UseExecuteContractArgs<Message>) => {
+  const { data: account } = useAccount();
+  const accountAddress = account?.bech32Address;
+
+  const mutationFn = (args: ExecuteContractMutationArgs<Message>) => {
+    const executeArgs: ExecuteContractArgs<Message> = {
+      ...args,
+      fee: args.fee ?? "auto",
+      senderAddress: accountAddress,
+      contractAddress,
+    };
+
+    return executeContract(executeArgs);
+  };
+
+  const queryKey = ["USE_EXECUTE_CONTRACT", onError, onLoading, onSuccess, contractAddress];
+  const mutation = useMutation(queryKey, mutationFn, {
+    onError: (err, data) => Promise.resolve(onError?.(err, data)),
+    onMutate: onLoading,
+    onSuccess: (txResponse) => Promise.resolve(onSuccess?.(txResponse)),
```

Comment:
to avoid confusion let's change the naming
```suggestion
    onSuccess: (executeResult) => Promise.resolve(onSuccess?.(executeResult)),
```